lib/components/os/resinos.js

```diff
@@ -71,3 +67,28 @@ exports.injectNetworkConfiguration = (image, configuration) => {
     path: '/system-connections/resin-wifi'
   }, wifiConfiguration.join('\n'))
 }
+
+module.exports = class ResinOS {
+  constructor (image, fetch, configuration) {
+    this.image = image
+    this.fetch = fetch
+    this.configuration = configuration
+  }
+
+  async runFetch () {
+    console.log(`Downloading device type OS into ${this.image}`)
+    await this.fetch(this.image)
+  }
+
+  async init () {
+    console.log(`Configuring resinOS image: ${this.image}`)
+    await injectResinConfiguration(this.image, this.configuration)
+    await injectNetworkConfiguration(this.image, this.configuration)
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  async sshHostOS (command, uuid, privateKeyPath) {
```

Comment:
Maybe we should call this just `.ssh()`? Apart from ResinOS, we don't have any OS with the concept of SSHing into the containers, so "host ssh" should be the default